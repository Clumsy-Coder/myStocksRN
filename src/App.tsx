import 'react-native-gesture-handler';
import React from 'react';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { PersistGate } from 'redux-persist/integration/react';

import store, { persistor } from 'src/redux/store';
import { DrawerNavigator } from 'src/routes/AppRoutes';

const App = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <NavigationContainer>
          <DrawerNavigator />
        </NavigationContainer>
      </PersistGate>
    </Provider>
  );
};

export default App;
