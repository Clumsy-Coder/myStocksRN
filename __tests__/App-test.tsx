import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import fs from 'fs';
import path from 'path';

import App from 'src/App';

describe('<App />', () => {
  const wrapper = shallow(<App />);

  it('Renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a <Provider />', () => {
    expect(wrapper.find('Provider')).toHaveLength(1);
  });
});

describe('Essential files', () => {
  const DIR_BASE = path.resolve(__dirname, '..'); // <project_root>

  it('jest.config.js file should exist', () => {
    const filepath = path.resolve(DIR_BASE, 'jest.config.js');
    expect(fs.existsSync(filepath)).toBe(true);
  });

  it('jest.setup.js file should exist', () => {
    const filepath = path.resolve(DIR_BASE, 'jest.setup.js');
    expect(fs.existsSync(filepath)).toBe(true);
  });

  it('tsconfig.json file should exist', () => {
    const filepath = path.resolve(DIR_BASE, 'tsconfig.json');
    expect(fs.existsSync(filepath)).toBe(true);
  });

  it('tsconfig.jest.json file should exist', () => {
    const filepath = path.resolve(DIR_BASE, 'tsconfig.jest.json');
    expect(fs.existsSync(filepath)).toBe(true);
  });

  it('babel.config.js file should exist', () => {
    const filepath = path.resolve(DIR_BASE, 'babel.config.js');
    expect(fs.existsSync(filepath)).toBe(true);
  });

  it('metro.config.js file should exist', () => {
    const filepath = path.resolve(DIR_BASE, 'metro.config.js');
    expect(fs.existsSync(filepath)).toBe(true);
  });
});
