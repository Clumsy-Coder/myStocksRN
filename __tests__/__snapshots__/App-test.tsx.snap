// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders as expected 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ForwardRef(NavigationContainer)>
      <Connect(HomeStackNavigator) />
    </ForwardRef(NavigationContainer)>
  </PersistGate>
</Provider>
`;
